openraft/src/core/admin.rs

```diff
@@ -204,6 +212,20 @@ impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> LeaderS
 
         tracing::debug!(?new_config, "new_config");
 
+        if let Err(e) = self.is_all_new_members_line_rate(new_members.cloned().collect::<BTreeSet<_>>(), blocking) {
+            let _ = tx.send(Err(e));
+            return Ok(());
+        }
+
+        self.append_membership_log(new_config, Some(tx)).await?;
+        Ok(())
+    }
+
+    fn is_all_new_members_line_rate(
+        &self,
+        new_members: BTreeSet<C::NodeId>,
+        blocking: bool,
+    ) -> Result<(), ClientWriteError<C>> {
```

Comment:
It seems it is possible to make `new_members` a `& BTreeSet`.
That would simplify things a bit.